eng/common/scripts/Package-Properties.ps1

```diff
@@ -120,38 +109,76 @@ function Get-AllPkgProperties ([string]$ServiceDirectory = $null)
-                }
+                $pkgPropsResult = Operate-OnPackages -ciFilePath $ciFile -ServiceDirectory $dir.Name -pkgPropsResult $pkgPropsResult
             }
         }
     } 
     else
     {
         $serviceDir = Join-Path $RepoRoot "sdk" $ServiceDirectory
-        if (Test-Path (Join-Path $serviceDir "ci.yml"))
-        {
-            $activePkgList = Get-PkgListFromYml -ciYmlPath (Join-Path $serviceDir "ci.yml")
-            if ($activePkgList -ne $null)
-            {
-                $pkgPropsResult = Operate-OnPackages -activePkgList $activePkgList -ServiceDirectory $ServiceDirectory -pkgPropsResult $pkgPropsResult
-            }
+        $ciFiles = fetchCiFiles -serviceDirectory $serviceDir -includeincludeMgmtPkg $includeMgmt
+        foreach ($ciFile in $ciFiles) {
+            $pkgPropsResult = Operate-OnPackages -ciFilePath $ciFile -ServiceDirectory $ServiceDirectory -pkgPropsResult $pkgPropsResult
         }
     }
 
     return $pkgPropsResult
 }
 
-function Operate-OnPackages ($activePkgList, $ServiceDirectory, [Array]$pkgPropsResult)
+function GetMetaData($lang){
+    switch ($lang) {
+      "java" {
+        $metadataUri = "https://raw.githubusercontent.com/Azure/azure-sdk/master/_data/releases/latest/java-packages.csv"
+        break
+      }
+      ".net" {
+        $metadataUri = "https://raw.githubusercontent.com/Azure/azure-sdk/master/_data/releases/latest/dotnet-packages.csv"
+        break
+      }
+      "python" {
+        $metadataUri = "https://raw.githubusercontent.com/Azure/azure-sdk/master/_data/releases/latest/python-packages.csv"
+        break
+      }
+      "javascript" {
+        $metadataUri = "https://raw.githubusercontent.com/Azure/azure-sdk/master/_data/releases/latest/js-packages.csv"
+        break
+      }
+      default {
+        Write-Host "Unrecognized Language: $language"
+        exit(1)
+      }
+    }
+  
+    $metadataResponse = Invoke-RestMethod -Uri $metadataUri -method "GET" -MaximumRetryCount 3 -RetryIntervalSec 10 | ConvertFrom-Csv
+  
+    return $metadataResponse
+}
+
+function fetchCiFiles([string]$serviceDirectory, [bool]$includeMgmtPkg) {
+    # We'd better to display artifacts based on the order of ci.yml, ci.data.yml and ci.mgmt.yml.
+    $ciFiles = @()
+    if (Test-Path (Join-Path $serviceDirectory ci.yml)) {
+        $ciFiles += (Join-Path $serviceDirectory ci.yml)
+    }
+    if (Test-Path (Join-Path $serviceDirectory ci.data.yml)) {
+        $ciFiles += (Join-Path $serviceDirectory ci.data.yml)
+    }
+    if ($includeMgmt -and (Test-Path (Join-Path $serviceDirectory ci.mgmt.yml))) {
+        $includeMgmtPkg += (Join-Path $serviceDirectory ci.mgmt.yml)
+    }
+    return $ciFiles
+}
+
+function Operate-OnPackages ($ciFilePath, $ServiceDirectory, [Array]$pkgPropsResult)
 {
-    foreach ($pkg in $activePkgList)
+    $activePkgList = Get-PkgListFromYml -ciYmlPath $ciFilePath
+    foreach ($pkg in $activePkgList.name)
```

Comment:
Is this correct? We aren't looping over the name property.

Reply:
The yaml parser is a little weird.
If we only have one artifact. Then the activePkgList we get looks like below:
```
activePkgList[0]: name
activePkgList[1]: groupid
activePkgList[2]: softname
```
It is working as expected when we have multiple artifacts.

Reply:
We just need to make sure what we return from the function is a array instead of an array or a single item.